### ISSUE TITLE
YellowBox text color should use black text to reach accessible contrast levels

### ISSUE THREAD

**Issue by fvsch** [facebook/react-native#24965]
Because developers need accessible tools too. :)

The main text style for YellowBox seems defined here:
https://github.com/facebook/react-native/blob/master/Libraries/YellowBox/UI/YellowBoxStyle.js#L26-L28

It's white with an opacity parameter, but in practice it seems to always be full or high opacity. Let's say it's always white, to make things simple.

The yellow background uses partial opacity. Against a dark and a white app background I'm getting those final backdrop colors:

- Black background: `#EDB12F`, [WCAG 2 contrast ratio = 1.92:1](https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=EDB12F)
- White background: `#F9BE3B`, [WCAG 2 contrast ratio = 1.68:1](https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=F9BE3B)

WCAG 2 considers that the minimum contrast should be 4.5:1 for normal text, and 3:1 for large text.

Switching the text color to black gives us contrast ratios in the 10:1 to 12:1 range, which is great. Some dark gray such as `#333333` might work too, though the contrast ratio takes a nosedive rather quickly for even slightly lighter grays.

For comparison, the red error display uses accessible color contrast ([5.4:1 in my tests](https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=D11A26)).

**Comment by fvsch** [comment]
Steps to reproduce:
1. Start a react-native project in dev mode in an iOS or Android emulator
2. Add `console.warn('Hello')` in your JS code, e.g. in the render method of your root React component

Here's a screenshot showing low contrast text in a YellowBox.

![screenshot](https://user-images.githubusercontent.com/243601/58010343-711d5c00-7af0-11e9-8c3a-96dc33b1f1dd.png)

**Comment by matthargett** [comment]
I actually heard from a sight-impaired React Native developer recently who complained that YellowBox ignored the font size preferences from the system settings. I love the idea of making React Native more accessible for app developers *and* their users.
